### PR TITLE
Update parent pom and core requirement

### PR DESCRIPTION
**Pull request by alecharp**
Kind of a follow up of jenkinsci/jenkins#5320.

ping @car-roll 

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -64,16 +64,16 @@
     <properties>
         <revision>2.22</revision>
         <changelist>-SNAPSHOT</changelist>
-        <jenkins.version>2.277.1</jenkins.version>
+        <jenkins.version>2.303.1</jenkins.version>
```
Comment (car-roll) [inline pom.xml:67]: 2.303 does seem like a big bump, but it does seem like the majority of users are on that.

**Review by car-roll — APPROVED** [review]

**Review by bitwiseman — APPROVED** [review]
The 2.303.x line is the current LTS line, so I'd generally rather not force people to it, but there aren't other changes to be released. This seems fine to me.

**Comment by car-roll** [comment]
Yikes, I accidentally bumped jenkins to 2.304 in #120, fixing here and reflecting change in logs

File: pom.xml (hunk 1 above)
Comment (jglick) [reply pom.xml:66]: FTR: https://github.com/jenkinsci/jenkins/pull/5320 is first in 2.303.x as an LTS line, so this is not arbitrary.

**Comment by jglick** [comment]
> Kind of a follow up of https://github.com/jenkinsci/jenkins/pull/5320.

@alecharp @car-roll can you explain in what sense this is a follow-up to that core change? This baseline is causing me some problems in handling #172 so I would like to know if it was truly necessary or if I can in fact revert the baseline to, say, 2.289.x at least for one release.

**Pull request merged**